connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -301,10 +300,20 @@ private void processConnectorConfigUpdates(Set<String> connectorConfigUpdates) {
             boolean remains = configState.contains(connectorName);
             log.info("Handling connector-only config update by {} connector {}",
                     remains ? "restarting" : "stopping", connectorName);
-            worker.stopConnector(connectorName);
+            if (worker.ownsConnector(connectorName)) {
+                worker.stopConnector(connectorName);
+            } else {
+                log.warn("Worker does not own connector {} so cannot stop it", connectorName);
+            }
             // The update may be a deletion, so verify we actually need to restart the connector
-            if (remains)
-                startConnector(connectorName);
+            if (remains) {
+                try {
+                    startConnector(connectorName);
+                } catch (Throwable t) {
+                    log.warn("Failed to start connector {}", connectorName, t);
+                    onFailure(connectorName, t);
```

Comment:
Until now, it looks like we isolated the status changes to `Worker`, which makes them work automatically for both standalone and distributed mode. This patch seems to address distributed mode, but can't most of the same errors happen in `StandaloneHerder` too? I think this is related to @hachikuji's comment below too -- basically if we can push some of this handling into Worker then I think we'd keep the use of the `onFailure` better localized. It might actually also reduce some code redundancy between the two Herder implementations.